packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -1334,3 +1351,77 @@ String _computeChecksum(Iterable<String> names, String getVersion(String name))
   }
   return ((upperCheck << 8) | lowerCheck).toRadixString(16).padLeft(4, '0');
 }
+
+/// Create a synthetic Flutter SDK so that pub version solving does not get
+/// stuck on the old versions.
+Directory createTemporaryFlutterSdk(FileSystem fileSystem, Directory realFlutter, List<PubspecYaml> pubspecs) {
```

Comment:
So this is filling in a fake Flutter SDK to trick pub into doing the right thing. How will we know if something changes with pub, and this no longer can trick it?

Reply:
I wouldn't say its tricking pub, its exposing the SDK the same way we do for `flutter packages get`, via FLUTTER_ROOT and the file system. If the implementation of pub changes enough to break this, everything else regarding dependency solving with the SDK will break too

![2020-03-27](https://user-images.githubusercontent.com/8975114/77807877-761f3600-7046-11ea-9dbf-ee1f396837ed.png)

Reply:
Fair enough